src/multitaper.jl

```diff
@@ -0,0 +1,518 @@
+Any keyword arguments accepted by [`MTConfig`](@ref) may be passed here, or an `MTConfig` object itself.
+"""
+function MTSpectrogramConfig{T}(n_samples, n_overlap_samples, mt_config::MTConfig{T}) where {T}
+    samples_per_window = mt_config.n_samples
+    fs = mt_config.fs
+    # Dimensions of the spectrogram power matrix
+    n_time_points = div(n_samples - samples_per_window,
+                        samples_per_window - n_overlap_samples) + 1
+
+    window_duration = samples_per_window / fs
+    overlap_duration = n_overlap_samples / fs
+
+    hop = window_duration - overlap_duration
+    time = range(window_duration / 2; step=hop, length=n_time_points)
+    return MTSpectrogramConfig{T,typeof(mt_config)}(n_samples, n_overlap_samples,
+                                                              n_time_points, time,
+                                                              mt_config)
+end
+
+function MTSpectrogramConfig{T}(n_samples, samples_per_window, n_overlap_samples;
+                                          fs=1, kwargs...) where {T}
+    mt_config = MTConfig{T}(samples_per_window; fs=fs, kwargs...)
+    return MTSpectrogramConfig{T}(n_samples, n_overlap_samples, mt_config)
+end
+
+"""
+    mt_spectrogram!(destination::AbstractMatrix,
+                                signal::AbstractVector,
+                                config::MTSpectrogramConfig) -> Spectrogram
+
+Computes a multitaper spectrogram using the parameters specified in `config`.
+
+* `destination`: `config.mt_config.n_freq_bins` x `config.n_time_points` matrix. This can be created by `allocate_output(config)`.
+* `signal`: vector of length `config.n_samples`
+* `config`: an [`MTSpectrogramConfig`](@ref) object to hold temporary variables and configuration settings.
+"""
+@views function mt_spectrogram!(destination::AbstractMatrix,
+                                          signal::AbstractVector,
+                                          config::MTSpectrogramConfig)
+    if size(destination) != (config.mt_config.n_freq_bins, config.n_time_points)
+        throw(DimensionMismatch("""Expected `destination` to be of size `(config.mt_config.n_freq_bins, config.n_time_points)`;
+                got `size(destination) == $(size(destination)) and
+                `(config.mt_config.n_freq_bins, config.n_time_points)` = $((config.mt_config.n_freq_bins, config.n_time_points))"""))
+    end
+    if length(signal) != config.n_samples
+        throw(DimensionMismatch("""Expected `signal` to be of length `config.n_samples`;
+                got `length(signal) = $(length(signal)) and `config.n_samples` = $(config.n_samples)"""))
+    end
+    samples_per_window = config.mt_config.n_samples
+    subepochs = arraysplit(signal, samples_per_window, config.n_overlap_samples)
+    for (time_index, subepoch) in enumerate(subepochs)
+        mt_pgram!(destination[:, time_index], subepoch, config.mt_config)
+    end
```

Comment:
five lines to do an in-place multitapered spectrogram! I think this kind of shows the point of layering the config objects, since the layout there kind of reflects how the code actually works, i.e. parameters are not mixed between `MTSpectrogramConfig` and `MTConfig` so hopefully it's clear what parameter is needed for what bit of the process.